### PR TITLE
NoDuplicateSemicolonsFixer - Remove overcomplete fixer

### PR DESCRIPTION
**Pull request by SpacePossum**
With the `NoEmptyStatementFixer` merged the `NoDuplicateSemicolonsFixer` has become overcomplete since it covers only a subset of the cases `NoEmptyStatementFixer` fixes.
I see no value in keeping the `NoDuplicateSemicolonsFixer` but maybe others do?

### CODE REVIEW

File: UPGRADE.md [hunk 1]
```diff
@@ -77,7 +77,7 @@ array_element_no_space_before_comma            | no_whitespace_before_comma_in_a
 array_element_white_space_after_comma          | whitespace_after_comma_in_array
 blankline_after_open_tag                       | blank_line_after_opening_tag
 double_arrow_multiline_whitespaces             | no_multiline_whitespace_around_double_arrow
-duplicate_semicolon                            | no_duplicate_semicolons
+duplicate_semicolon                            | no_empty_statement
```
Comment (keradus) [inline UPGRADE.md]: `no empty statement` does more, right? please add 3rd column

File: tests/FixerFactoryTest.php [hunk 2]
```diff
@@ -251,7 +246,6 @@ public function getFixersPriorityCases()
             array($fixers['combine_consecutive_unsets'], $fixers['no_whitespace_in_blank_lines']), // tested also in: combine_consecutive_unsets,no_whitespace_in_blank_lines.test
             array($fixers['combine_consecutive_unsets'], $fixers['no_trailing_whitespace']), // tested also in: combine_consecutive_unsets,no_trailing_whitespace.test
             array($fixers['combine_consecutive_unsets'], $fixers['no_extra_consecutive_blank_lines']), // tested also in: combine_consecutive_unsets,no_extra_consecutive_blank_lines.test
-            array($fixers['no_duplicate_semicolons'], $fixers['combine_consecutive_unsets']), // tested also in: no_duplicate_semicolons,combine_consecutive_unsets.test
```
Comment (keradus) [inline tests/FixerFactoryTest.php]: no need to replace all of those with no_empty_statement ?

File: src/Fixer/Semicolon/NoDuplicateSemicolonsFixer.php [hunk 3]
```diff
@@ -1,72 +0,0 @@
...
-    {
-        return 'Remove duplicated semicolons.';
-    }
-
-    /**
-     * {@inheritdoc}
-     */
-    public function getPriority()
-    {
-        // should be run before the BracesFixer, NoSinglelineWhitespaceBeforeSemicolonsFixer, NoMultilineWhitespaceBeforeSemicolonsFixer, SwitchCaseSemicolonToColonFixer and NoUselessReturnFixer.
```
Comment (keradus) [inline src/Fixer/Semicolon/NoDuplicateSemicolonsFixer.php]: no need to move it to `no_empty_statement` ?

**Comment by keradus** [comment]
:+1: for idea ;)

File: tests/Fixtures/Integration/priority/no_duplicate_semicolons,switch_case_semicolon_to_colon.test [hunk 4]
```diff
@@ -1,17 +0,0 @@
---TEST--
-Integration of fixers: no_duplicate_semicolons,switch_case_semicolon_to_colon.
```
Comment (stof) [inline tests/Fixtures/Integration/priority/no_duplicate_semicolons,switch_case_semicolon_to_colon.test]: shoudln't this be converted to testing the integration of ``switch_case_semicolon_to_colon`` with the new fixer ?

Comment (SpacePossum) [inline tests/Fixtures/Integration/priority/no_duplicate_semicolons,switch_case_semicolon_to_colon.test]: has been done;
https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/master/tests/Fixtures/Integration/priority/no_empty_statement,switch_case_semicolon_to_colon.test

this one `tests/Fixtures/Integration/priority/no_duplicate_semicolons,combine_consecutive_unsets.test` does need a replacement indeed :+1:

File: src/Fixer/Semicolon/NoDuplicateSemicolonsFixer.php (hunk 3 above)
Comment (SpacePossum) [inline src/Fixer/Semicolon/NoDuplicateSemicolonsFixer.php]: indeed and the comment there was out dated, all upped.

File: src/Fixer/Semicolon/NoEmptyStatementFixer.php [hunk 5]
```diff
@@ -73,7 +73,8 @@ public function getDescription()
      */
     public function getPriority()
     {
-        // should be run before the BracesFixer, SpacesBeforeSemicolonFixer, SpacesAfterSemicolonFixer, WhitespacyLinesFixer, MultilineSpacesBeforeSemicolonFixer, CombineConsecutiveUnsetsFixer and ExtraEmptyLinesFixer.
+
+        // should be run before the BracesFixer, CombineConsecutiveUnsetsFixer, NoExtraConsecutiveBlankLinesFixer, NoMultilineWhitespaceBeforeSemicolonsFixer, NoSinglelineWhitespaceBeforeSemicolonsFixer, NoTrailingCommaInListCallFixer, NoUselessReturnFixer, NoWhitespaceInBlankLinesFixer, SpaceAfterSemicolonFixer, SwitchCaseSemicolonToColonFixer.
```
Comment (keradus) [inline src/Fixer/Semicolon/NoEmptyStatementFixer.php]: please split to multiline

Comment (GrahamCampbell) [inline src/Fixer/Semicolon/NoEmptyStatementFixer.php]: also, a newline before this comment shouldn't have been added

Comment (SpacePossum) [inline src/Fixer/Semicolon/NoEmptyStatementFixer.php]: :+1:  done

**Comment by keradus** [comment]
almost ready to merge, @SpacePossum please rebase

**Comment by GrahamCampbell** [comment]
:+1:

**Comment by SpacePossum** [comment]
rebased :)

**Comment by keradus** [comment]
Thank you @SpacePossum.

**Pull request merged**